src/System.Buffers.Experimental/System/Buffers/Text/BufferWriter_strings.cs

```diff
@@ -0,0 +1,200 @@
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Utf8;
+
+namespace System.Buffers.Text
+{
+    public ref partial struct BufferWriter
+    {
+        #region String
+        public bool TryWrite(string text)
+        {
+            if (!TryWrite(text, out var written)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void Write(string text)
+        {
+            while (!TryWrite(text)) Resize();
+        }
+
+        public bool TryWrite(string text, TransformationFormat format)
+        {
+            int written;
+            if (!TryWrite(text, out written)) return false;
+            if (!format.TryTransform(Free, ref written)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void Write(string text, TransformationFormat format)
+        {
+            while (!TryWrite(text, format)) Resize();
+        }
+
+        public bool TryWriteLine(string text)
+        {
+            if (!TryWriteLine(text, out var written)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void WriteLine(string text)
+        {
+            while (!TryWriteLine(text)) Resize();
+        }
+
+        public bool TryWriteLine(string text, TransformationFormat format)
+        {
+            int written;
+            if (!TryWriteLine(text, out written)) return false;
+            if (!format.TryTransform(Free, ref written)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void WriteLine(string text, TransformationFormat format)
+        {
+            while (!TryWriteLine(text, format)) Resize();
+        }
+        #endregion
+
+        #region UTF8String
+        public bool TryWrite(Utf8Span text)
+        {
+            int written = text.Bytes.Length;
+            if (!text.Bytes.TryCopyTo(Free)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void Write(Utf8Span text)
+        {
+            while (!TryWrite(text)) Resize();
+        }
+
+        public bool TryWrite(Utf8Span text, TransformationFormat format)
+        {
+            int written = text.Bytes.Length;
+            if (!text.Bytes.TryCopyTo(Free)) return false;
+            if (!format.TryTransform(Free, ref written)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void Write(Utf8Span text, TransformationFormat transformation)
+        {
+            while (!TryWrite(text, transformation)) Resize();
+        }
+
+        public bool TryWriteLine(Utf8Span text)
+        {
+            if (!TryWriteLine(text, out var written)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void WriteLine(Utf8Span text)
+        {
+            while (!TryWriteLine(text)) Resize();
```

Comment:
Could this ever lead to a non-terminating loop?

Reply:
I dont think so, the Resize method throws when the buffer does not actually resize